### PR TITLE
Adds README for jib-gradle-plugin.

### PR DESCRIPTION
**Pull request by coollog**
First part of #208 .

~~This should be merged after https://discuss.gradle.org/t/cannot-resolve-artifacts-for-custom-plugin-published-to-the-plugin-portal/26496 is resolved.~~ **RESOLVED**

Preview at https://github.com/google/jib/tree/gradle-readme/jib-gradle-plugin

### CODE REVIEW

**Comment by coollog** [comment]
@parkjam4 Hey Jimmy, could you take a pass over this?

File: README.md [hunk 1]
```diff
@@ -22,7 +22,6 @@ Jib is a Maven plugin for building Docker and OCI images for your Java applicati
 
 These features are not currently supported but will be added in later releases.
```
Comment (parkjam4) [inline README.md]: Would be nice to add additional bullets here.
But, if you don't have any additional features planned, should just combine these into one sentence.

File: jib-gradle-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+These features are not currently supported but will be added in later releases.
+
+* Support for WAR format
+* Define credentials in configuration
+
+## Quickstart
+
+### Setup
+
+*Make sure you are using Gradle version 4.6 or newer.*
```
Comment (parkjam4) [inline jib-gradle-plugin/README.md]: Change to:
Make sure you are using Gradle version 4.6 or later.

**Comment by coollog** [comment]
@loosebazooka @TadCordle @parkjam4 good to go?

File: jib-gradle-plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+```groovy
+jib {
+
+  // Configures the base image to build your application on top of.
+  // OPTIONAL
+  from {
+    // The image reference for the base image.
+    // OPTIONAL STRING, defaults to 'gcr.io/distroless/java'
+    image =
```
[the remark this answers is not in the document]
Comment (loosebazooka) [reply jib-gradle-plugin/README.md]: Hah, I have no recommendations. I guess nested types have their own table with entries?

| Parameter        | Type           | Description  |
| ------------- |-------------| -----|
| from *(optional)*     | [Image-Reference](linktoimagerefrence) | The base image for your application... |
| to      | [Image-Reference](linktoimagerefrence)     | Target Image |
| jvmFlags | list      | flags to bake into the image |

Image Reference

| Parameter        | Type           | Description  |
| ------------- |-------------| -----|
| image    | String | image reference |
| credentialHelper  | String | crendential helper suffix |

[the remark this answers is not in the document]
Comment (loosebazooka) [reply jib-gradle-plugin/README.md]: But from and to appear to have different descriptions for Image-Reference, I don't really know what you'd do there.

**Comment by coollog** [comment]
That was the format I'd used for the Maven documentation (https://github.com/google/jib#extended-usage). I added the required to the to directly. How does it look?

**Pull request merged**